staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -407,11 +389,14 @@ func (o *ApplyOptions) Run() error {
 				Force:        &o.ForceConflicts,
 				FieldManager: o.FieldManager,
 			}
+
+			helper := resource.NewHelper(info.Client, info.Mapping)
 			if o.ServerDryRun {
-				options.DryRun = []string{metav1.DryRunAll}
+				if err := resource.VerifyDryRun(info.Mapping.GroupVersionKind, o.DynamicClient, o.DiscoveryClient); err != nil {
+					helper = helper.DryRun(o.ServerDryRun)
```

Comment:
I don't know if you need to re-assign the helper? Isn't it changing it in place?

Reply:
Oh right it changes in place, I'll remove this assignment